google-cloud-pubsub/lib/google/cloud/pubsub/received_message.rb

```diff
@@ -63,6 +63,49 @@ def ack_id
           @grpc.ack_id
         end
 
+        ##
+        # Returns the delivery attempt counter for the message. The delivery attempt
+        # counter is `1 + (the sum of number of NACKs and number of
+        # ack_deadline exceeds)` for the message.
```

Comment:
@hongalex Should this description be updated to mention that the value may be `0` if a dead letter policy is not set?

Reply:
Yeah that would be ideal. We might be looking to change this later but for now including that's good.

Reply:
Done